lib/HLSL/DxilPromoteResourcePasses.cpp

```diff
@@ -223,3 +233,464 @@ ModulePass *llvm::createDxilPromoteStaticResources() {
 INITIALIZE_PASS(DxilPromoteStaticResources,
                 "hlsl-dxil-promote-static-resources",
                 "DXIL promote static resource use", false, false)
+
+// Mutate resource into handle.
```

Comment:
It would be useful to add to the comment a little more context for the pass, like:
```
// Mutate high-level resource type into handle.
// This is used for SM 6.6+, on libraries only, where
// CreateHandleForLib is eliminated, and high-level resource
// types are only preserved in metadata for reflection purposes.
```

Reply:
Actually, one comment block incorporating this and the brief description below would be helpful.  You can base it off of this if you think it is correct:
```
// Mutate high-level resource type into handle.
// This is used for SM 6.6+, on libraries only, where
// CreateHandleForLib is eliminated, and high-level resource
// types are only preserved in metadata for reflection purposes.
// Overview
// 1. collectCandidates - collect to MutateValSet
//    a) Start from resource global variable, function parameter/ret, alloca.
//    b) Propagate to all insts, GEP/ld/st/phi/select/called functions.
// 2. mutateCandidates
//    a) Mutate all non-function value types.
//    b) Mutate functions by creating new function with new type, then
//       splice original function blocks into new function, and
//       replace old argument uses with new function's arguments.
```